engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java

```diff
@@ -0,0 +1,401 @@
+import io.camunda.zeebe.engine.processing.streamprocessor.StreamProcessorListener;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedEventRegistry;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.CommandResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.LegacyTypedStreamWriter;
+import io.camunda.zeebe.logstreams.log.LogStreamBatchWriter;
+import io.camunda.zeebe.logstreams.log.LogStreamReader;
+import io.camunda.zeebe.logstreams.log.LoggedEvent;
+import io.camunda.zeebe.logstreams.storage.LogStorage;
+import io.camunda.zeebe.logstreams.util.ListLogStorage;
+import io.camunda.zeebe.logstreams.util.SyncLogStream;
+import io.camunda.zeebe.logstreams.util.SynchronousLogStream;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.scheduler.Actor;
+import io.camunda.zeebe.scheduler.ActorScheduler;
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.streamprocessor.StreamProcessor;
+import io.camunda.zeebe.streamprocessor.StreamProcessor.Phase;
+import io.camunda.zeebe.streamprocessor.StreamProcessorMode;
+import io.camunda.zeebe.util.FileUtil;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.slf4j.Logger;
+
+public final class StreamPlatform {
+
+  private static final String SNAPSHOT_FOLDER = "snapshot";
+  private static final Map<Class<?>, ValueType> VALUE_TYPES = new HashMap<>();
+  private static final Logger LOG = Loggers.STREAM_PROCESSING;
+  private static final int DEFAULT_PARTITION = 1;
+  private static final String STREAM_NAME = "stream-";
+
+  static {
+    TypedEventRegistry.EVENT_REGISTRY.forEach((v, c) -> VALUE_TYPES.put(c, v));
+  }
+
+  private final Path dataDirectory;
+  private final List<AutoCloseable> closeables;
+  private final ActorScheduler actorScheduler;
+  private final CommandResponseWriter mockCommandResponseWriter;
+  private final StreamProcessorListener mockStreamProcessorListener;
+  private final Map<String, LogContext> logContextMap = new HashMap<>();
+  private final Map<String, ProcessorContext> streamContextMap = new HashMap<>();
+  private boolean snapshotWasTaken = false;
+  private final StreamProcessorMode streamProcessorMode = StreamProcessorMode.PROCESSING;
+  private RecordProcessor recordProcessor;
+
+  private final WriteActor writeActor = new WriteActor();
+  private final ZeebeDbFactory zeebeDbFactory;
+
+  public StreamPlatform(
```

Comment:
Most parts are from TestStream and StreamProcessingComposite, I tried to clean up it already a bit